logging/tests/system.py

```diff
@@ -123,6 +124,31 @@ def tearDown(self):
     def _logger_name():
         return 'system-tests-logger' + unique_resource_id('-')
 
+    def test_list_entry_with_unregistered(self):
+        from google.protobuf import any_pb2
+        from google.protobuf import descriptor_pool
+        from google.cloud.logging import entries
+
+        pool = descriptor_pool.Default()
+        type_name = 'google.cloud.audit.AuditLog'
+        # Make sure the descriptor is not known in the registry.
+        with self.assertRaises(KeyError):
+            pool.FindMessageTypeByName(type_name)
+
+        type_url = 'type.googleapis.com/' + type_name
+        filter_ = self.TYPE_FILTER.format(type_url)
+        entry_iter = iter(
+            Config.CLIENT.list_entries(page_size=1, filter_=filter_))
+        protobuf_entry = next(entry_iter)
```

Comment:
@lukesneeringer @tseaver 

1. What sort of `@Retry` should I bake into this `next()` call on the iterator?
1. I realized that **after** this `next(entry_iter)` keeps going, which surprised me. We should probably have a `max_results` keyword as well?